<a href="https://colab.research.google.com/github/meetdelvadiya/Appliance-Energy-Prediction/blob/main/Appliance_Energy_Prediction_Project_Meet_Delvadiya.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Data-driven prediction of energy use of appliances**

The data set is at 10 min for about 4.5 months. The house temperature and humidity conditions
were monitored with a ZigBee wireless sensor network. Each wireless node transmitted the
temperature and humidity conditions around 3.3 min. Then, the wireless data was averaged for
10 minutes periods. The energy data was logged every 10 minutes with m-bus energy meters.
Weather from the nearest airport weather station (Chievres Airport, Belgium) was downloaded
from a public data set from Reliable Prognosis (rp5.ru) and merged together with the
experimental data sets using the date and time column. Two random variables have been
included in the data set for testing the regression models and to filter out non-predictive attributes
(parameters).

In [ ]:
# importing libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

from datetime import datetime
import time
import dateutil

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import PolynomialFeatures
from sklearn.ensemble import RandomForestRegressor
import xgboost as xgb
from sklearn.tree import DecisionTreeRegressor
from sklearn.metrics import r2_score
from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

In [ ]:
# mounting drive
from google.colab import drive
 
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
dataset=pd.read_csv("/content/drive/MyDrive/Dataset/data_application_energy.csv")

In [ ]:
dataset.shape

(19735, 29)

In [ ]:
#It gives first 5 rows of dataset
dataset.head()

date  Appliances  lights  ...  Tdewpoint        rv1        rv2
0  2016-01-11 17:00:00          60      30  ...        5.3  13.275433  13.275433
1  2016-01-11 17:10:00          60      30  ...        5.2  18.606195  18.606195
2  2016-01-11 17:20:00          50      30  ...        5.1  28.642668  28.642668
3  2016-01-11 17:30:00          50      40  ...        5.0  45.410389  45.410389
4  2016-01-11 17:40:00          60      40  ...        4.9  10.084097  10.084097

[5 rows x 29 columns]

In [ ]:
#It gives last 5 rows of dataset
dataset.tail()

date  Appliances  lights  ...  Tdewpoint        rv1        rv2
19730  2016-05-27 17:20:00         100       0  ...  13.333333  43.096812  43.096812
19731  2016-05-27 17:30:00          90       0  ...  13.300000  49.282940  49.282940
19732  2016-05-27 17:40:00         270      10  ...  13.266667  29.199117  29.199117
19733  2016-05-27 17:50:00         420      10  ...  13.233333   6.322784   6.322784
19734  2016-05-27 18:00:00         430      10  ...  13.200000  34.118851  34.118851

[5 rows x 29 columns]

In [ ]:
#It gives total columns, data types and null count of dataset.
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19735 entries, 0 to 19734
Data columns (total 29 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   date         19735 non-null  object 
 1   Appliances   19735 non-null  int64  
 2   lights       19735 non-null  int64  
 3   T1           19735 non-null  float64
 4   RH_1         19735 non-null  float64
 5   T2           19735 non-null  float64
 6   RH_2         19735 non-null  float64
 7   T3           19735 non-null  float64
 8   RH_3         19735 non-null  float64
 9   T4           19735 non-null  float64
 10  RH_4         19735 non-null  float64
 11  T5           19735 non-null  float64
 12  RH_5         19735 non-null  float64
 13  T6           19735 non-null  float64
 14  RH_6         19735 non-null  float64
 15  T7           19735 non-null  float64
 16  RH_7         19735 non-null  float64
 17  T8           19735 non-null  float64
 18  RH_8         19735 non-null  float64
 19  T9  

In [ ]:
# table to return datatype,null and unique counnt
def printinfo():
    temp = pd.DataFrame(index=dataset.columns)
    temp['data_type'] = dataset.dtypes
    temp['null_count'] = dataset.isnull().sum()
    temp['unique_count'] = dataset.nunique()
    return temp

printinfo()

data_type  null_count  unique_count
date           object           0         19735
Appliances      int64           0            92
lights          int64           0             8
T1            float64           0           722
RH_1          float64           0          2547
T2            float64           0          1650
RH_2          float64           0          3376
T3            float64           0          1426
RH_3          float64           0          2618
T4            float64           0          1390
RH_4          float64           0          2987
T5            float64           0          2263
RH_5          float64           0          7571
T6            float64           0          4446
RH_6          float64           0          9709
T7            float64           0          1955
RH_7          float64           0          5891
T8            float64           0          2228
RH_8          float64           0          6649
T9            float64           0           924
RH_9          float64           0          3388
T_out         float64           0          1730
Press_mm_hg   float64           0          2189
RH_out        float64           0           566
Windspeed     float64           0           189
Visibility    float64           0           413
Tdewpoint     float64           0          1409
rv1           float64           0         19735
rv2           float64           0         19735

As shown above , there are no null values in the dataset

In [ ]:
dataset["exact_date"]=dataset['date'].str.split(' ').str[0]
dataset["days"]=(dataset['date'].str.split(' ').str[0])
dataset["days"]=(dataset['days'].apply(dateutil.parser.parse, dayfirst=True))
dataset["days"]=(dataset['days'].dt.day_name())

In [ ]:
dataset['lights'].value_counts()

0     15252
10     2212
20     1624
30      559
40       77
50        9
60        1
70        1
Name: lights, dtype: int64

There are many '0' values in 'lights' column so we will drop it. 
We are separating 'date' column into 'exact date' and 'days' so we will drop original 'date' column.



In [ ]:
dataset.drop(['date','lights'],axis=1,inplace=True)

In [ ]:
dataset.head()

Appliances     T1       RH_1  ...        rv2  exact_date     days
0          60  19.89  47.596667  ...  13.275433  2016-01-11  Tuesday
1          60  19.89  46.693333  ...  18.606195  2016-01-11  Tuesday
2          50  19.89  46.300000  ...  28.642668  2016-01-11  Tuesday
3          50  19.89  46.066667  ...  45.410389  2016-01-11  Tuesday
4          60  19.89  46.333333  ...  10.084097  2016-01-11  Tuesday

[5 rows x 29 columns]

In [ ]:
#It gives some basic statistical details like percentile, mean, std, max etc.
dataset.describe()

Appliances            T1  ...           rv1           rv2
count  19735.000000  19735.000000  ...  19735.000000  19735.000000
mean      97.694958     21.686571  ...     24.988033     24.988033
std      102.524891      1.606066  ...     14.496634     14.496634
min       10.000000     16.790000  ...      0.005322      0.005322
25%       50.000000     20.760000  ...     12.497889     12.497889
50%       60.000000     21.600000  ...     24.897653     24.897653
75%      100.000000     22.600000  ...     37.583769     37.583769
max     1080.000000     26.260000  ...     49.996530     49.996530

[8 rows x 27 columns]

Here, Appliances have mean value around 100 and minimum value is 10 whereas maximum is 1080. So we can ignore outliers.